src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,263 @@
+
+        return genotypes;
+    }
+
+    private static final class AnnotationManager {
+
+        private final List<String> annotationGroupsToUse;
+        private final List<String> annotationsToUse;
+
+        private AnnotationManager(final List<String> annotationGroupsToUse, final List<String> annotationsToUse){
+            this.annotationGroupsToUse = annotationGroupsToUse;
+            this.annotationsToUse = annotationsToUse;
+
+            final Set<String> unknownAnnots = Sets.difference(new HashSet<>(annotationsToUse), new HashSet<>(AnnotationManager.getAllAnnotationNames()));
+            if (!unknownAnnots.isEmpty()){
+                throw new BadArgumentValue("annotation", "Annotations " + unknownAnnots + " were not found; please check that you have specified the annotation name correctly");
+            }
+
+            final Set<String> unknownGroups =  Sets.difference(new HashSet<>(annotationGroupsToUse), new HashSet<>(AnnotationManager.getAllAnnotationGroupNames()));
+            if (!unknownGroups.isEmpty()){
+                throw new BadArgumentValue("group", "Unknown annotation group in " + unknownGroups + ". Known groups are " + Arrays.toString(AnnotationGroup.values()) );
+            }
+
+        }
+
+        static AnnotationManager ofSelected(final List<String> annotationGroupsToUse, final List<String> annotationsToUse, final List<String> annotationsToExclude){
+            final List<String> groups = new ArrayList<>(annotationGroupsToUse);//make copy
+            final List<String> annots = new ArrayList<>(annotationsToUse);//make copy
+            annots.removeAll(annotationsToExclude);
+            return new AnnotationManager(groups, annots);
+        }
+
+        static AnnotationManager ofAll(final List<String> annotationsToExclude){
+            final List<String> groups = getAllAnnotationGroupNames();
+            final List<String> annots = getAllAnnotationNames();
+            annots.removeAll(annotationsToExclude);
+            return new AnnotationManager(groups, annots);
+        }
+
+        private static List<String> getAllAnnotationNames() {
+            final Set<VariantAnnotation> union = Sets.union(new HashSet<>(makeAllGenotypeAnnotations()), new HashSet<>(AnnotationManager.makeAllInfoFieldAnnotations()));
+            return union.stream().map(a -> a.getClass().getSimpleName()).collect(Collectors.toList());
+        }
+
+        public static List<String> getAllAnnotationGroupNames() {
+            return Stream.of(AnnotationGroup.values()).map(v -> v.name()).collect(Collectors.toList());
+        }
+
+        public List<InfoFieldAnnotation> createInfoFieldAnnotations() {
+            final List<InfoFieldAnnotation> all = makeAllInfoFieldAnnotations();
+            return filterAnnotations(all);
+        }
+
+        private static List<InfoFieldAnnotation> makeAllInfoFieldAnnotations() {
+            return Arrays.asList(
+                    new InbreedingCoeff(Standard, ActiveRegionBased),
```

Comment:
`InbreedingCoeff` can optionally be constructed with a Set of `founderIds` -- what mechanism is there for the engine to get extra information like this into an annotation instance? In GATK3 the mechanism was an `initialize()` method, which had the downside of decoupling construction from initialization.